exporter/datadogexporter/internal/metrics/consumer.go

```diff
@@ -54,28 +54,33 @@ func (c *Consumer) toDataType(dt metrics.DataType) (out datadogV2.MetricIntakeTy
 }
 
 // runningMetrics gets the running metrics for the exporter.
-func (c *Consumer) runningMetrics(timestamp uint64, buildInfo component.BuildInfo) (series []datadogV2.MetricSeries) {
+func (c *Consumer) runningMetrics(timestamp uint64, buildInfo component.BuildInfo, metadata metrics.RuntimeMetricsTelemetry) (series []datadogV2.MetricSeries) {
 	for host := range c.seenHosts {
 		// Report the host as running
-		runningMetric := DefaultMetrics("metrics", host, timestamp, buildInfo)
+		runningMetric := DefaultMetrics("metrics", host, timestamp, buildInfo, "")
 		series = append(series, runningMetric...)
 	}
 
 	for tag := range c.seenTags {
-		runningMetrics := DefaultMetrics("metrics", "", timestamp, buildInfo)
+		runningMetrics := DefaultMetrics("metrics", "", timestamp, buildInfo, "")
```

Comment:
Having the empty string as an argument isn't a pretty API and makes it hard to understand what is missing without actually going to the definition of `DefaultMetrics`. You simply see an empty string, but aren't exactly sure what wasn't provided. Instead, I suggest improving this function by changing its signature to this:
```go
func DefaultMetrics(exporterType string, hostname string, timestamp uint64, tags []string) []datadogV2.MetricSeries
```
A slice of string tags is quite commonly used in metrics code. You could create a helper method in the `internal/metrics` package to avoid boilerplate:
```go
func TagsFromBuildInfo(bi component.BuildInfo) []string {
    return []string{"version:"+bi.Version, "command:"+bi.Command}
}
```
Then, you could call it like this:
```go
metrics.DefaultMetrics("traces", host, uint64(now), metrics.TagsFromBuildInfo(buildInfo))
```
or:
```go
tags := append(metrics.TagsFromBuildInfo(buildInfo), "language:"+lang")
metrics.DefaultMetrics("traces", host, uint64(now), tags)
```
This way, you haven't added an argument to a function that is only non-empty 10% of the time, rendering it "mostly" useless. Then, the code becomes only slightly more complex during the exception (when lang. is needed), and not the other way around.

WDYT?

Reply:
That pattern makes a lot of sense, thanks for the suggestion!